txtl/sbmlutil.py

```diff
@@ -0,0 +1,230 @@
+    unitdef.setId('square_metre')
+    unit = unitdef.createUnit()
+    unit.setKind(libsbml.UNIT_KIND_METRE)
+    unit.setExponent(2)
+    unit.setScale(0)
+    unit.setMultiplier(1)
+
+    # Set up required units and containers
+    model.setTimeUnits('second')            # set model-wide time units
+    model.setExtentUnits('mole')            # set model units of extent
+    model.setSubstanceUnits('mole')         # set model substance units
+    model.setLengthUnits('metre')           # area units (never used?)
+    model.setAreaUnits('square_metre')      # area units (never used?)
+    model.setVolumeUnits('litre')           # default volume unit
+
+    # Define the default compartment
+    compartment = model.createCompartment()
+    compartment.setId('txtl')
+    compartment.setConstant(True)           # keep compartment size constant
+    compartment.setSpatialDimensions(3)     # 3 dimensional compartment
+    compartment.setVolume(1e-6)             # 1 microliter
+
+    return document, model, compartment
+
+# Helper function to add a species to the model
+def add_species(mixture, type, name, ic=None, debug=False):
+    model = mixture.model   # Get the model where we will store results
+    
+    # Construct the species name
+    prefix = type + " " if type != None else ""
+    species_name = prefix + name
+    
+    # Construct the species ID
+    species_id = re.sub(" ", "_", species_name)
+    species_id = re.sub("--", "_", species_id)
+    species_id = re.sub(":", "_", species_id)
+    
+    # Check to see if this species is already present
+    species = find_species(model, species_id)
+    if species == None:
+        if debug: print("Adding species %s" % species_name)
+        species = model.createSpecies()
+        prefix = type + " " if type != None else ""
+        species.setName(species_name)
+        species.setId(species_id)
+        species.setCompartment(mixture.compartment.getId())
+        species.setConstant(False)
+        species.setBoundaryCondition(False)
+        species.setHasOnlySubstanceUnits(False) #! TODO: confirm this is OK
+
+    else:
+        if debug: print("add_species: %s already exists", species.getId())
+        
+    # Set the initial concentration (if specified)
+    #! TODO: Decide whether to warn if species is already present
+    #! TODO: add initial concentrations if reaction is present
+    if ic != None:
+        if debug: print("    %s IC = %s" % (species_name, ic))
+        species.setInitialConcentration(float(ic))
+
+    return species
+
+# Look for a species in the current mixture
+def find_species(mixture, species_name):
+    model = mixture.model   # Get the model where we will store results
+    
+    # Construct the species ID (no-op if already a species ID)
+    species_id = re.sub(" ", "_", species_name)
+    species_id = re.sub("--", "_", species_id)
+    species_id = re.sub(":", "_", species_id)
+    
+    return model.getSpecies(species_id)
```

Comment:
There is a libSBML function called `model.getElementBySId(id)` which might be helpful to have one general function for finding species, parameters, reactions etc. It could serve a similar purpose as `sbioselect` in MATLAB Simbiology.

Reply:
Is there a way to get only species (rather than reactions or something else)?

Reply:
I think what you have already used works fine, if you only want to get species, i.e. `model.getSpecies(id)`. 
What I was suggesting was to maybe provide another function (while possibly keeping this `get_species()`  function as it is) which could get all other component with a given id in a mixture.